### PR TITLE
Update sample descriptor schemas

### PR DESCRIPTION
**Pull request by jkokosar**
* Update sample descriptor schema
* Update NGS reads descriptor schema
* Add microarray data descriptor schema
* Remove obsolete descriptor schemas

### CODE REVIEW

File: docs/CHANGELOG.rst [hunk 1]
```diff
@@ -24,13 +19,16 @@ Changed
 * Update differential expression (DE) processors
 * Update ``generate_diffexpr_cuffdiff`` django-admin command
 * Save gene_id source to ``output.source`` for DE, expression and related objects
+* Update sample descriptor schema
 
 Added
 -----
 * Process definition documentation style and text improvements
+* Add tests to ensure generators produce the same results
```
Comment (nkran) [inline docs/CHANGELOG.rst]: Are test supposed to be in this commit? I can't find them. I ran the generators manually and they seem to work.

Comment (jkokosar) [inline docs/CHANGELOG.rst]: This is unrelated to the descriptor schemas. This change in to fix an error in the change-log that is present in the current master. There were two "Added" sections introduced by previous merge.

File: resolwe_bio/descriptors/sample.yml [hunk 2]
```diff
@@ -276,46 +133,54 @@
...
+        - name: fragmentation_method
+          label: Fragmentation method
           type: basic:string
-          required: false
         - name: ip_tag
           label: IP Tag
           type: basic:string
           required: false
           hidden: "geo.experiment_type != 'ChIP-Seq'"
         - name: sample_control
```
Comment (mstajdohar) [inline resolwe_bio/descriptors/sample.yml]: Rename to sample_type.

**Comment by tjanez** [comment]
@jkokosar, please rebase this on #119, which should work around platform differences when testing the generators.
